rosie/src/main/java/com/karumi/rosie/domain/usecase/RosieUseCase.java

```diff
@@ -74,7 +76,9 @@ protected void notifySuccess(Object... values) {
    */
 
   protected void notifyError(final Error error) throws ErrorNotHandledException {
-    onErrorCallback.get().onError(error);
+    if (onErrorCallback.get() != null) {
+      onErrorCallback.get().onError(error);
```

Comment:
Hi, @Aracem checking if the callback is null is the way to go, but there is a small detail related to the current implementation we should change.

As this code is being used in an asynchronous context we should check always if the same variable is null and not invoking the ``get()`` method over the weak reference when checking the nullity.

Extracting ``onSuccessCallback.get()`` and ``onErrorCallback.get()`` into two variables will avoid a possible race condition where the garbage collector could release the reference just after checking ``if != null`` and before using the reference.

Reply:
You are totally right. Give me a second to change it. Sorry

Reply:
Changes ammended @pedrovgs